CefSharp.OffScreen/ChromiumWebBrowser.cs

```diff
@@ -614,20 +678,77 @@ void IRenderWebBrowser.InvokeRenderAsync(BitmapInfo bitmapInfo)
         /// in it's lock scope.
         /// </summary>
         /// <param name="bitmapInfo">information about the bitmap to be rendered</param>
+        /// <exception cref="InvalidOperationException">When background image is null when on blend mode.</exception>
         public virtual void InvokeRenderAsync(BitmapInfo bitmapInfo)
         {
             var gdiBitmapInfo = (GdiBitmapInfo)bitmapInfo;
-            if (bitmapInfo.CreateNewBitmap)
+
+            if (BlendingConfiguration == PopupBlending.Separate)
             {
-                if (Bitmap != null)
+                if (bitmapInfo.CreateNewBitmap)
                 {
-                    Bitmap.Dispose();
-                    Bitmap = null;
+                    if (Bitmap != null)
+                    {
+                        Bitmap.Dispose();
+                        Bitmap = null;
+                    }
+
+                    Bitmap = gdiBitmapInfo.CreateBitmap();
                 }
+            }
+            else if (BlendingConfiguration == PopupBlending.Blend)
+            {
+                if (bitmapInfo.CreateNewBitmap || PopupJustOpened)
+                {
+                    if (Bitmap != null)
+                    {
+                        Bitmap.Dispose();
+                        Bitmap = null;
+                    }
 
-                Bitmap = gdiBitmapInfo.CreateBitmap();
+                    if (gdiBitmapInfo.IsPopup)
+                    {
+                        if (PopupImage != null)
+                        {
+                            PopupImage.Dispose();
+                            PopupImage = null;
+                        }
+                        PopupImage = gdiBitmapInfo.CreateBitmap();
+                        PopupJustOpened = false;
+                    }
+                    else
+                    {
+                        if (BackGroundImage != null)
+                        {
+                            BackGroundImage.Dispose();
+                            BackGroundImage = null;
+                        }
+
+                        BackGroundImage = gdiBitmapInfo.CreateBitmap();
+                    }
+                }
+
+                if (PopupImage != null && BackGroundImage != null)
+                {
+                    Bitmap = MergeBackGroundPopupBitmaps();
+                }
+                else if (PopupImage == null)
+                {
+                    Bitmap = (Bitmap)BackGroundImage.Clone();
+                }
+                else if (BackGroundImage == null)
+                {
+                    throw new InvalidOperationException("Background image was empty. Should not happen in Blend mode.");
+                }
             }
+            TriggerNewScreenshot();
```

Comment:
Method that would override InvokeRenderAsync are not able to trigger the new screenshot event (since C# only allows to fire an event from its class). I thought it could be missing, so I created this method, but maybe I misunderstood something about this event ?